lib/jobs/generate-tag.js

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = generateTagJobFactory;
+di.annotate(generateTagJobFactory, new di.Provide('Job.Catalog.GenerateTag'));
+di.annotate(generateTagJobFactory, new di.Inject(
+    'Job.Base',
+    'Services.Waterline',
+    'Protocol.Events',
+    'JobUtils.CatalogSearchHelpers',
+    'anchor',
+    'Logger',
+    'Util',
+    'Promise',
+    'Assert',
+    '_'
+));
+function generateTagJobFactory(
+    BaseJob,
+    waterline,
+    eventsProtocol,
+    catalogSearch,
+    anchor,
+    Logger,
+    util,
+    Promise,
+    assert,
+    _
+) {
+
+    var logger = Logger.initialize(generateTagJobFactory);
+
+    /**
+     *
+     * @param {Object} [options]
+     * @constructor
+     */
+    function GenerateTagJob(options, context, taskId) {
+        GenerateTagJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = context.target || options.nodeId;
+        assert.isMongoId(this.nodeId, 'context.target || options.nodeId');
+    }
+
+    util.inherits(GenerateTagJob, BaseJob);
+
+    var RULE_DELIMITER = /\./g;
+
+    /**
+     * @memberOf GenerateTagJob
+     * @returns {Promise}
+     */
+    GenerateTagJob.prototype._run = function run() {
+        var self = this;
+
+        waterline.tags.find({}).then(function (tags) {
+            var catalogTypes = _(tags)
```

Comment:
I just discovered this usage pattern of lodash a couple weeks ago, glad you are using it.